### PR TITLE
Work on failing tests

### PR DESCRIPTION
**Pull request by jamesagnew**

### CODE REVIEW

File: hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/search/PersistedJpaBundleProvider.java [hunk 1]
```diff
@@ -292,10 +292,15 @@ public Integer size() {
 		SearchCoordinatorSvcImpl.verifySearchHasntFailedOrThrowInternalErrorException(mySearchEntity);
 
 		Integer size = mySearchEntity.getTotalCount();
-		if (size == null) {
+		if (size != null) {
+			return Math.max(0, size);
```
Comment (fil512) [inline hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/search/PersistedJpaBundleProvider.java]: size can be negative??

File: hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/search/PersistedJpaBundleProvider.java [hunk 2]
```diff
@@ -292,10 +292,15 @@ public Integer size() {
...
+		}
+
+		if (mySearchEntity.getSearchType() == SearchTypeEnum.HISTORY) {
 			return null;
+		} else {
+			return mySearchCoordinatorSvc.getSearchTotalFromRunningSearchIfExists(myUuid).orElse(null);
```
Comment (fil512) [inline hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/search/PersistedJpaBundleProvider.java]: I like the orElse syntax on Optional return values.  It reads really well.  I tend to prefix those methods with Optional even though it makes them even longer...

File: hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/search/SearchCoordinatorSvcImpl.java [hunk 3]
```diff
@@ -231,10 +250,17 @@ public void cancelAllActiveSearches() {
...
 		}
 
+		ourLog.trace("Finished looping");
 
 		List<Long> pids = mySearchResultCacheSvc.fetchResultPids(search, theFrom, theTo);
 
+		ourLog.trace("Fetched {} results", pids.size());
+
+		if (pids.size() < theTo - theFrom) {
+			if (search.getNumFound() >= theTo) {
```
Comment (fil512) [inline hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/search/SearchCoordinatorSvcImpl.java]: empty if statements?

File: hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/search/SearchCoordinatorSvcImpl.java [hunk 4]
```diff
@@ -549,6 +598,28 @@ protected BaseTask(Search theSearch, IDao theCallingDao, SearchParameterMap theP
...
+		 */
+		Integer awaitInitialSync() {
+			ourLog.trace("Awaiting initial sync");
+			do {
+				try {
+					if (getInitialCollectionLatch().await(250, TimeUnit.MILLISECONDS)) {
+						break;
+					}
+				} catch (InterruptedException e) {
+					// Shouldn't happen
```
Comment (fil512) [inline hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/search/SearchCoordinatorSvcImpl.java]: Thread.currentThread().interrupt();

File: hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/search/PersistedJpaBundleProvider.java (hunk 1 above)
Comment (jamesagnew) [reply hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/search/PersistedJpaBundleProvider.java]: Honestly, I don't know why that is there... and i'm scared to touch it :(

File: hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/search/PersistedJpaBundleProvider.java (hunk 2 above)
Comment (jamesagnew) [reply hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/search/PersistedJpaBundleProvider.java]: Agree

File: hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/search/SearchCoordinatorSvcImpl.java (hunk 3 above)
Comment (jamesagnew) [reply hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/search/SearchCoordinatorSvcImpl.java]: Oops, that had a debug log in it. Removing.

File: hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/search/SearchCoordinatorSvcImpl.java (hunk 4 above)
Comment (jamesagnew) [reply hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/search/SearchCoordinatorSvcImpl.java]: done

File: hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/search/PersistedJpaBundleProvider.java (hunk 1 above)
Comment (fil512) [reply hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/search/PersistedJpaBundleProvider.java]: You just added it...?

Comment (jamesagnew) [reply hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/search/PersistedJpaBundleProvider.java]: No it was already there... I just move it up a bit: https://github.com/jamesagnew/hapi-fhir/pull/1510/files/0724db1a92e7141a90f73d2468c231d92e2e022d#diff-ed4963ecb631eb0419df5b6d79905d92L298

**Pull request merged**